backend/ingest_approved_drugs.py

```
@@ -0,0 +1,427 @@
+    if not cols:
+        raise RuntimeError(
+            "molecules table has none of the expected columns "
+            "(chembl_id, name, smiles, molecular_weight, is_approved)"
+        )
+
+    print("🧹 Truncating molecules table…")
+    cur.execute("DELETE FROM molecules")
+
+    col_names = ", ".join(c for c, _ in cols)
+    placeholders = ", ".join("?" * len(cols))
+    sql = f"INSERT INTO molecules ({col_names}) VALUES ({placeholders})"
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)